# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студенты гр. 0382 Корсунов Антон и Здобнова Ксения. Вариант №11

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

**Интервальной** называется статистическая оценка, которая задается двумя числами - концами интервала (в котором может находиться значение параметра $θ$
****
**Точностью $θ^*$** - статистической оценки параметра $θ$ называется значение $δ$, удовлетворяющее неравенству:
$$δ > |θ^* - θ| (1)$$
Чем меньше значение $δ$, тем точнее оценка $θ$
****
**Надежностью (доверительной вероятностью)** оценки $θ^*$ называется вероятность $γ$, с которой выполняется неравенство (1):
$$γ = P\{δ > |θ^* - θ|\} (2)$$
или
$$γ = P\{θ^* - δ < θ < θ^* + δ\} (3)$$

Выражения (2) и (3) следует понимать в том смысле, что интервал
$$(θ^* - δ ; θ^* + δ) (4)$$
с вероятностью (надежностью) $γ$ содержит в себе (покрывает) истинное значение оцениваемого параметра $θ$
****
Интервал (4) называется **доверительным интервалом** и является интервальной оценкой параметра $θ$

Существует 2 вида интервальных оценок: **односторонние и двусторонние**. При двусторонней оценке задаются обе границы доверительного интервала, зависящие от доверительной вероятности $γ$: левая $β_1(γ)$ и правая $β_2(γ)$:
$$P\{β_1(γ) < θ < β_2(γ)\} = p$$
Для одностороннего доверительного интеравала границы интервалов задаются так, чтобы:
$$P\{β(γ) > θ\} =γ$$
или
$$P\{β(γ) < θ\} =γ$$
****
Значение $q = 1 - γ$ называют **уровнем значимости**. Значение γ задают обычно равным 0.95, 0.99, 0.995, 0.999
****
**Статистической гипотезой** называют гипотезу о виде неизвестного распределения исследуемой случайной величины или о значении параметров известных распределений.
****
Выдвинутую гипотезу называют **нулевой или основной гипотезой** и обозначают $H_0$
****
**Конкурирующей или альтернативной** называют гипотезу $H_1$, которая противоречит основной гипотезе.
****
Различают **простые гипотезы** (содержат только одно предположение) и **сложные гипотезы** (содержат более одного предположения)
****
**Ошибка первого рода** состоит в том, что нулевая гипотеза отвергается, хотя она верна
****
**Ошибка второго рода** состоит в том, что нулевая гипотеза принимается, хотя она неверна
****
Вероятность совершения ошибки первого рода называют **уровнем значимости** и обозначают $α$. Как правило, уровень значимости назначают равным от 0.001 до 0.2
****
Величину $1 - β$ - вероятность того, что гипотеза отвергается, когда она верна, называют **мощностью критерия**

Следует отметить, что принятие гипотезы не означает ее справедливость, а лишь констатирует, что, в соответствии с используемым критерием и уровнем значимости, гипотеза согласуется с данными наблюдений.
****
Для проверки статистических гипотез предлагаются различные критерии, которые представляют собой значения случайных величин, имеющих те или иные известные законы распределения.

При проверке статистических гипотез на основе выборочных данных вычисляются значения этих случайных величин (**наблюденные значения**), которые сопоставляются с некоторыми, так называемыми, **критическими** (зависящими от заданного уровня значимости $α$) **значениями** этих случайных величин. На основании этого сопоставления принимается решение о принятии или не принятии нулевой гипотезы.

****
**Областью принятия нулевой гипотезы** называют совокупность значения критерия, при которых нулевую гипотезу принимают
****
**Критической областью** называют совокупность значений критерия, при которых нулевую гипотезу отвергают
****
Область принятия гипотезы и критическая область разделены точками, которые называются **критическими значениями критерия** $K_{кр}$
****
Область непринятия гипотезы может быть односторонней (правосторонней или левосторонней) и двусторонней

**Правосторонней** называют критическую область, определяемую неравенством $K_{набл} > K_{кр}$

**Левосторонней** называют критическую область, определяемую неравенством $K_{набл} < K_{кр}$

**Двусторонней** называют критическую область, определяемую неравенствами $K_{набл} < K_{кр_1}$ и $K_{набл} > K_{кр_2}$

****
**Критерий Пирсона (или критерий $χ^2$)** - правосторонний критерий. Согласно этому критерию на основании выборочных данных вычисляется "наблюденное" значение случайной величины $χ^2$:
$$χ^2_{набл} = \sum^K_{i=1} \frac{(n_i - n_i')^2}{n_i'} (5)$$
Здесь $n_i'$ - теоретические частоты - частоты, с которыми нормально распределенная величина с параметрами $\overline{x_в}$ и $S$ попала бы в $i$-й интервал интервального ряда при проведении $N$ испытаний
****
Распределение хи-квадрат зависит от числа свободы $k$, которое вычисляется по формуле (для нормального распределения):
$$k = K - 3 (6)$$

По числу степеней свободы и уровню значимости вычисляется значение $χ^2_{крит} = χ^2(α,k)$. Это значение можно определить с помощью соответствующей таблицы.

Область принятия гипотезы $H_0$ определяется условием:
$$χ^2_{набл} ≤ χ^2_{крит} (7)$$

## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $\chi^2$. Дать содержательную интерпретацию полученным результатам.

## Выполнение работы


**Пункт 1**

**Вычислить точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности $γ∈\{0.95,0.99\}$. Сделать выводы.**

Имеется выборка объемом $N$ = 108:

Рассматривается случайная величина $t$:
$$t = \frac{\overline{x}_в - a}{s / \sqrt{N}}$$

$\overline{x}_в$ - выборочное среднее

$a$ - математическое ожидание

$s$ - исправленное СКО

$N$ - объем выборки

Эта случайная величина распределена по закону $S(t, N)$ - закону Стьюдента с $k = N - 1$ степенями свободы

Поскольку функция $S(t, N)$ является четной, справедливо соотношение:
$$P(|\frac{\overline{x}_в - a}{s / \sqrt{N}}|<t_γ) = 2 ∫_0^{t_γ}S(t, N)dt = γ (8)$$

$$(8) → P(\overline{x}_в - t_γs/\sqrt{N} < a < \overline{x}_в + t_γs/\sqrt{N}) = γ$$

В результате требуемый доверительный интервал, покрывающий неизвестное значение параметра $a$ с надежностью $γ$ принимает следующий вид:
$$(\overline{x}_в - t_γs/\sqrt{N} ; \overline{x}_в + t_γs/\sqrt{N})$$
Точность $δ$ = $t_γs/\sqrt{N}$

Значение $t_γ$ определяется по заданным значениям $γ$ и $N$ по соответствующей таблице

Пусть $γ = 0.95$, тогда значение $t_γ$ согласно таблице (при $N = 108$) равно $1.984$

Пусть $γ = 0.99$, тогда значение $t_γ$ согласно таблице (при $N = 108$) равно $2.627$

In [53]:
import math #подключение библиотек
from prettytable import PrettyTable

In [107]:
x_mean = 1635.172839506173
s = 286.0932746391938
t_gamma = 1.984
N = 108

def get_interval_1(x_mean, s, t_gamma, N):
  d = t_gamma * s / math.sqrt(N)
  left_border =  x_mean - d
  right_border = x_mean + d

  return d, left_border, right_border

temp = get_interval_1(x_mean, s, t_gamma, N)
print("при надежности 0.95:")
print(temp[0])
print("(" + str(temp[1]) + ";" + str(temp[2]) + ")")

t_gamma = 2.627
temp = get_interval_1(x_mean, s, t_gamma, N)
print("\nпри надежности 0.99:")
print(temp[0])
print("(" + str(temp[1]) + ";" + str(temp[2]) + ")")

при надежности 0.95:
54.61820696442328
(1580.5546325417497;1689.7910464705963)

при надежности 0.99:
72.31957141912295
(1562.85326808705;1707.4924109252959)


То есть, согласно вычислениям, при надежности $0.95$ **точность** равна $54.61820696442328$, а **доверительный интервал** $(1580.5546325417497;1689.7910464705963)$

А при при надежности $0.99$ **точность** равна $72.31957141912295$, а **доверительный интервал** $(1562.85326808705;1707.4924109252959)$

**Выводы по пункту:**

Согласно проведенным вычислением истинное значение математического ожидания содержится в интервале $(1580.5546325417497;1689.7910464705963)
$ с вероятностью $0.95$ и в интервале $(1562.85326808705;1707.4924109252959)$ с вероятностью $0.99$.

Стоит отметить, что хоть и второй интервал с большей доверительной вероятностью покрывает истинное значение искомого параметра, он менее точен, чем первый (т.к. точность для него численно больше, чем у первого). Если посмотреть на полученную ранее формулу для определения точности $δ$ = $t_γs/\sqrt{N}$, то улучшить точность, не уменьшая при этом доверительную вероятность, можно путем увеличения объема выборки.

**Пункт 2**

**Для вычисления границ доверительного интервала для среднеквадратичного отклонения определить значение $q$ при заданных $γ$ и $n$. Построить доверительные интервалы, сделать выводы.**

Рассматривается соотношение $P(|s-σ|<δ)=γ$ и равносильное ему:
$$P(s-δ<σ<s+δ)=γ$$
Преобразуется двойное неравенство $s-δ<σ<s+δ$ к виду:
$$s(1-δ/s)<σ<s(1+δ/s)$$
Вводится обозначение: $q = δ/s$, тогда:
$$s(1-q)<σ<s(1+q)$$
Предполагая $q<1:$
$$\frac{1}{s(1+q)}<\frac{1}{σ}<\frac{1}{s(1-q)}$$
Если умножить это неравенство на $s\sqrt{N-1}$:
$$\frac{\sqrt{N-1}}{(1+q)}<\frac{s\sqrt{N-1}}{σ}<\frac{\sqrt{N-1}}{(1-q)}$$
Далее пусть случайная величина $χ$ равна:
$$χ = \frac{s\sqrt{N-1}}{σ}$$
Функция плотности вероятности $R(χ, N)$ распределения случайной величины $χ$ зависит только от $N$
Тогда предыдущее двойное неравенство перепиштся в виде:
$$\frac{\sqrt{N-1}}{(1+q)}<χ<\frac{\sqrt{N-1}}{(1-q)}$$

Чтобы вероятность выполнения двойного неравенства равнялась, $γ$ необходимо выполнения условия:
$$∫_{\sqrt{N-1}/(1+q)}^{\sqrt{N-1}/(1-q)}R(χ, N)dχ = γ$$

Значение $q$, при котором, при заданных значениях $γ$ и $N$, предыдущее уравнение выполняется, определяется по соответствующей таблице.

В результате требуемый доверительный интервал ($s(1-q), s(1+q)$), покрывающий неизвестное значение параметра $σ$ с надежностью $γ$, будет построен

Пусть $γ = 0.95$, тогда значение $q$ согласно таблице (при $N = 108$) равно $0.143$

Пусть $γ = 0.99$, тогда значение $q$ согласно таблице (при $N = 108$) равно $0.198$

In [108]:
s = 286.0932746391938
q = 0.143

def get_interval_2(s, q):
  left_border =  s*(1-q)
  right_border = s*(1+q)

  return left_border, right_border

temp = get_interval_2(s, q)
print("при надежности 0.95:")
print("(" + str(temp[0]) + ";" + str(temp[1]) + ")")

q = 0.198
temp = get_interval_2(s, q)
print("\nпри надежности 0.99:")
print("(" + str(temp[0]) + ";" + str(temp[1]) + ")")

при надежности 0.95:
(245.18193636578908;327.00461291259853)

при надежности 0.99:
(229.44680626063342;342.73974301775417)


То есть, согласно вычислениям, при надежности $0.95$ **значение $q$** равно $0.143$, а **доверительный интервал** $(245.18193636578908;327.00461291259853)$

А при при надежности $0.99$ **значение $q$** равно $0.198$, а **доверительный интервал** $(229.44680626063342;342.73974301775417)$

**Выводы по пункту:**

Согласно проведенным вычислением истинное значение среднеквадратичного отклонения содержится в интервале $(245.18193636578908;327.00461291259853)$ с вероятностью $0.95$ и в интервале $(229.44680626063342;342.73974301775417)$ с вероятностью $0.99$.

Опять же, доверительный интервал во втором случаее менее точен (т.к. $δ = qs)$, но с большей вероятностью покрывает истинное значение искомого параметра

**Пункт 3**

**Проверить гипотезу о нормальности заданного распределения с помощью критерия $\chi^2$ (Пирсона). Для этого необходимо найти теоретические частоты и вычислить наблюдаемое значение критерия. Для удобства вычисления необходимо заполнить табл. 1.**

Выдвигается гипотеза $H_0$ - выборочные данные представляют значения случайной величины, распределенной по нормальному закону распределения. В качестве критерия проверки гипотезы будет использоваться критерий Пирсона (5):
$$χ^2_{набл} = \sum^K_{i=1} \frac{(n_i - n_i')^2}{n_i'} $$

$n_i'$ - теоретические частоты - частоты, с которыми нормально распределенная величина с параметрами $\overline{x_в}$ и $s$ попала бы в $i$-й интервал интервального ряда при проведении $N$ испытаний




Теоретические частоты вычисляются по следующему алгоритму:


1.   Пересчитываются значения границ интервалов интервального ряда. При этом левая граница первого интервала назначается равной $-∞$ ($z_1 = -∞)$, а правая граница последнего интервала назначается равной $+∞$ ($z_{K+1} = +∞)$. Остальные значения вычисляются по формуле: $$z_i = \frac{x_i - \overline{x_в}}{s}, i = 2, 3, ..., K$$
2.   Вычисляется вероятность попадания случайной величины в кадждый $i$-й интервал интервального ряда: $$p_i = Ф(z_{i+1}) - Ф(z_{i}), i = 1, 2, 3, ..., K$$ $Ф(z)$ - функция Лапласа: $Ф(z) = \frac{1}{\sqrt{2π}}∫_0^ze^{-\frac{t^2}{2}}dt$. Значения этой функции можно взять из соответствующей таблицы. При этом нужно учитывать нечетность этой функции и то, что $Ф(-∞) = -0.5$; $Ф(+∞) = 0.5$
3. Вычислить значения теоретических частот: $$n_i' = Np_i, i = 1, 2, 3, ..., K$$





In [109]:
# пересчитывание границ
K = 9
x_mean = 1635.172839506173
s = 286.0932746391938

z = [-math.inf]
interval_borders = [1040.0, 1190.888888888889, 1341.7777777777778,
                    1492.6666666666665, 1643.5555555555557, 1794.4444444444443,
                    1945.3333333333333, 2096.222222222222, 2247.1111111111113,
                    2398.0]
for i in range(1, K):
  z.append((interval_borders[i] - x_mean)/s)

z.append(math.inf)

for i in range(len(z)):
  print('z_' + str(i+1) + " = " + str(z[i]))

z_1 = -inf
z_2 = -1.552933920511037
z_3 = -1.0255224003374774
z_4 = -0.4981108801639184
z_5 = 0.029300640009642075
z_6 = 0.556712160183201
z_7 = 1.0841236803567607
z_8 = 1.6115352005303203
z_9 = 2.138946720703881
z_10 = inf


In [110]:
F = [-0.5, -0.4394, -0.3485, -0.1915, 0.012, 0.2123, 0.3599, 0.4463, 0.4898, 0.5]

print('Значения функции Лапласа согласно таблице:')

for i in range(len(F)):
  print('Ф(z_' + str(i+1) + ") = " + str(F[i]))

Значения функции Лапласа согласно таблице:
Ф(z_1) = -0.5
Ф(z_2) = -0.4394
Ф(z_3) = -0.3485
Ф(z_4) = -0.1915
Ф(z_5) = 0.012
Ф(z_6) = 0.2123
Ф(z_7) = 0.3599
Ф(z_8) = 0.4463
Ф(z_9) = 0.4898
Ф(z_10) = 0.5


In [113]:
p = []
for i in range(0, K):
  p.append(F[i+1] - F[i])

print('вероятность попадания случайной величины в кадждый  i -й интервал интервального ряда:')
for i in range(len(p)):
  print('p_' + str(i+1) + " = " + str(p[i]))

вероятность попадания случайной величины в кадждый  i -й интервал интервального ряда:
p_1 = 0.06059999999999999
p_2 = 0.09090000000000004
p_3 = 0.15699999999999997
p_4 = 0.20350000000000001
p_5 = 0.20029999999999998
p_6 = 0.1476
p_7 = 0.08639999999999998
p_8 = 0.04350000000000004
p_9 = 0.010199999999999987


In [116]:
N = 108
n_i_teor = []
for i in range(0, K):
  n_i_teor.append(N * p[i])

print('Теоретические частоты:')
for i in range(len(n_i_teor)):
  print("n'_i" + str(i+1) + " = " + str(n_i_teor[i]))

Теоретические частоты:
n'_i1 = 6.544799999999999
n'_i2 = 9.817200000000003
n'_i3 = 16.955999999999996
n'_i4 = 21.978
n'_i5 = 21.632399999999997
n'_i6 = 15.940800000000001
n'_i7 = 9.331199999999997
n'_i8 = 4.698000000000004
n'_i9 = 1.1015999999999986


Далее производятся вычисления для наблюденного $χ^2_{набл}$

In [121]:
n_i = [5, 11, 22, 20, 19, 14, 10, 5, 2] # абсолютные частоты попадания интервального ряда

fifth_column = []
sixth_column = []
seventh_column = []
eighth_column = []
for i in range(len(n_i)):
  temp1 = (n_i[i] - n_i_teor[i]) ** 2
  temp2 = n_i[i] ** 2


  fifth_column.append(temp1)
  sixth_column.append(temp1 / n_i_teor[i])
  seventh_column.append(temp2)
  eighth_column.append(temp2 / n_i_teor[i])

chi_square = sum(sixth_column)

print(chi_square)

3.542274342038122


То есть, $χ^2_{набл} = 3.542274342038122$

Вышеописанные вычисления в таблице:

In [122]:
myTable = PrettyTable(["i", "(x_{i-1}, x_i]","n_i", "p_i", "n'_i", "(n_i-n'i)^2",
    "(n_i-n'i)^2 / n'_i", "n^2_i", "n^2_i / n'_i"])

for i in range(0, K):
  myTable.add_row([i+1, 0, n_i[i], p[i], n_i_teor[i], fifth_column[i],
                   sixth_column[i], seventh_column[i], eighth_column[i]])

myTable.add_row(['sum', '-', sum(n_i), sum(p), sum(n_i_teor), '-',
                 sum(sixth_column), '-', sum(eighth_column)])

print(myTable)

+-----+----------------+-----+----------------------+--------------------+---------------------+----------------------+-------+--------------------+
|  i  | (x_{i-1}, x_i] | n_i |         p_i          |        n'_i        |     (n_i-n'i)^2     |  (n_i-n'i)^2 / n'_i  | n^2_i |    n^2_i / n'_i    |
+-----+----------------+-----+----------------------+--------------------+---------------------+----------------------+-------+--------------------+
|  1  |       0        |  5  | 0.06059999999999999  | 6.544799999999999  |  2.386407039999996  |  0.3646264270871526  |   25  | 3.819826427087154  |
|  2  |       0        |  11 | 0.09090000000000004  | 9.817200000000003  |  1.3990158399999923 |  0.1425066047345467  |  121  | 12.325306604734543 |
|  3  |       0        |  22 | 0.15699999999999997  | 16.955999999999996 |  25.44193600000004  |  1.5004680349138975  |  484  |  28.5444680349139  |
|  4  |       0        |  20 | 0.20350000000000001  |       21.978       |  3.9124840000000063 |  0.178018

Распределение хи-квадрат зависит от числа свободы $k$, которое вычисляется по формуле (для нормального распределения) (6):
$$k = K - 3$$

По числу степеней свободы и уровню значимости вычисляется значение $χ^2_{крит} = χ^2(α,k)$. Это значение можно определить с помощью соответствующей таблицы.

Область принятия гипотезы $H_0$ определяется условием (7):
$$χ^2_{набл} ≤ χ^2_{крит} $$

Пусть уровень значимости $\alpha = 0.05$

Число степеней свободы $k = K - 3 = 9 - 3 = 6$

Тогда согласно таблице $χ^2_{крит} = 2.45 $

Получается, что $χ^2_{набл} > χ^2_{крит} $, а значит нулевая гипотеза отвергается

**Пункт 4**

**Доказать, что $$\chi^2_{\textit{набл}} = \sum_{i = 1}^k\frac{n^2_i}{n'_i} - N.$$ Проконтролировать корректность вычисления $\chi^2_{\textit{набл}}$**

Согласно материалам лекций: $$χ^2_{набл} = \sum^K_{i=1} \frac{(n_i - n_i')^2}{n_i'} $$

В этой формуле можно раскрыть квадрат разности в числителе и сократить все получившиеся члены на знаменатель:

$$χ^2_{набл} = \sum^K_{i=1} \frac{(n_i - n_i')^2}{n_i'}  = \sum^K_{i=1} \frac{n_i^2 - 2 n_i n_i' + n_i'^2}{n_i'} = \sum^K_{i=1}\frac{n_i^2}{n_i'} - \sum^K_{i=1} \frac{ 2 n_i n_i'}{n_i'} + \sum^K_{i=1} \frac{ n_i'^2}{n_i'} = \sum^K_{i=1}\frac{n_i^2}{n_i'} - 2\sum^K_{i=1} n_i + \sum^K_{i=1} n_i'$$

В силу того, что $\sum^K_{i=1} n_i = \sum^K_{i=1} n_i' = N$, предыдущее выражение преобразуется в: $$\chi^2_{\textit{набл}} = \sum_{i = 1}^k\frac{n^2_i}{n'_i} - N.$$


Контроль вычислений $\chi^2_{\textit{набл}}$ с помощью доказанной выше формулы (левый член разности в формуле уже был вычислен ранее - самая нижняя ячейка последнего столбца в таблице 1), из него остается только вычесть объем выборки:

In [123]:
print(sum(eighth_column) - N)

3.5422743420381266


Как можно заметить, результаты совпали, контроль был произведен

**Пункт 5**

**По заданному уровню значимости $\alpha = 0.05$ и числу степеней свободы $df$ найти критическую точку $\chi^2_{\textit{крит}}$ и сравнить с наблюдаемым значением.**

$\chi^2_{\textit{крит}}$ была вычислена в **пункте 4**, значение оказалась меньше наблюденного, из-за чего нулевая гипотеза была отвергнута.

**Выводы по пункту:**

Проверка нулевой гипотезы о нормальности заданного распределения с помощью критерия $\chi^2$ (Пирсона) показала, что гипотезу следует отвергнуть. Это не означает, что заданное распределение не является нормальным, однако в соответствии с используемым критерием $K$ и уровнем значимости $\alpha = 0.05$, гипотеза не согласуется с данными наблюдений.

## Выводы

Работа выполнена на языке программирования Python. В ходе решения
поставленных задач были усвоены такие понятия как интервальная статистическая оценка, точность статистической оценки параметра, доверительная вероятность, доверительный интервал, критерий Пирсона. При нахождения доверительных интервалов был сделан вывод, что одним из способов улучшить точность интервала, не уменьшая доверительную вероятность, является увеличение объема выборки. При проверки статистической гипотезы о нормальности заданного распределения с помощью критерия Пирсона выяснилось, что в соответствии с выбранными критерием и уровнем значимости, гипотеза не согласуется с данными наблюдений. Также было доказано равенство двух формул для определения $χ^2_{набл}$.


